sw/device/freestanding/limits.h

```diff
@@ -0,0 +1,51 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_FREESTANDING_LIMITS_H_
+#define SW_DEVICE_FREESTANDING_LIMITS_H_
+
+#include "sw/device/freestanding/stdint.h"
+
+/**
+ * This header implements the limits.h standard header, as required by C11 S4.6.
+ * This header is specified in detail in S7.10 and S5.2.4.2.1 of the same.
+ */
+
+#define CHAR_BIT (8)
+#define SCHAR_MIN INT8_MIN
+#define SCHAR_MAX INT8_MAX
+#define UCHAR_MAX UINT8_MAX
+
+// NOTE: S5.2.5.2.2 specifies that the following defines should corresond to the
+// signedness of |char|; the RISCV ABI defines |char| to be unsigned. See
+// https://github.com/riscv/riscv-elf-psabi-doc/blob/master/riscv-elf.md#c-type-representation
+#define CHAR_MIN (0)
+#define CHAR_MAX UCHAR_MAX
+#define MB_LEN_MAX (1)
+
+// NOTE: the RISCV psABI defines the following widths for the integer types:
+// - short: 2
+// - int: 4
+// - long: <machine word>
+// - long long: 8
+// Compare the Rust types i16, i32, isize, and u64.
+// See
+// https://github.com/riscv/riscv-elf-psabi-doc/blob/master/riscv-elf.md#c-types
```

Comment:
I see why you put this here, but I feel like it should go in `stdint.h`.

Reply:
Sorry, this did end up in `stdint,h` I just hadn't gotten there yet.  Ignore this.

Reply:
This is just here to explain why MINs and MAXes of C primitives are given in terms of stdint limits.